compose/ui/ui/src/skikoMain/kotlin/androidx/compose/ui/platform/PlatformContextTextInputService.kt

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package androidx.compose.ui.platform
+
+import androidx.compose.ui.geometry.Rect
+import androidx.compose.ui.graphics.Matrix
+import androidx.compose.ui.text.TextLayoutResult
+import androidx.compose.ui.text.input.EditCommand
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.ImeOptions
+import androidx.compose.ui.text.input.OffsetMapping
+import androidx.compose.ui.text.input.PlatformTextInputService
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.ui.text.input.TextInputService
+import androidx.compose.ui.text.input.TextInputSession
+
+
+/**
+ * Platform specific text input service.
+ *
+ * This is a non-deprecated version of [PlatformTextInputService], which is needed because we can't
+ * expose deprecated APIs in [PlatformContext].
+ */
+interface PlatformContextTextInputService {
```

Comment:
I tried to undeprecate it [here](https://github.com/JetBrains/compose-multiplatform-core/commit/940c5c9643b2e92e83da96aa2d1872e9b06ae657) and realised, that its deprecation was justified. The old `PlatformTextInputService` manipulates immutable text field states and copies the whole text field instead of mutating it. The new `BasicTextField(TextFieldState)` on the other hand is designed that it just changes some portion in some data structure (probably a gap buffer).

So, to implement it properly, we have to design the new API that doesn't use immutable states.

Reply:
I pushed removing new public API. @MatkovIvan, please check.

We can merge it as is without optimisations in 1.7, and then do them in 1.8 (if needed)